### PR TITLE
[green][metric/1] collect metric of weekly green

### PR DESCRIPTION
**Pull request by can-anyscale**
Create a script to continuously query the number of weekly-release-blocker tag. This will be used as a main source to measure the readiness of weekly green.

Test:
- CI

```
bazel run //ci/ray_ci/automation:weekly_green_metric
[INFO 2024-01-22 21:59:36,085] credentials.py: 1124  Found credentials in environment variables.
[INFO 2024-01-22 21:59:37,098] weekly_green_metric.py: 29  Found 18 release blockers
```

### CODE REVIEW

File: ci/ray_ci/automation/weekly_green_metric.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    blockers = _get_release_blockers(github, repo)
+    logger.info(f"Found {blockers.totalCount} release blockers")
+
+
+def _get_release_blockers(
+    github: github.Github, repo: github.Repository.Repository
+) -> List[github.Issue.Issue]:
+    release_blocker_label = repo.get_label(WEEKLY_RELEASE_BLOCKER_TAG)
+
+    return github.get_user().get_issues(state="open", labels=[release_blocker_label])
```
Comment (aslonnie) [inline ci/ray_ci/automation/weekly_green_metric.py:34]: can we move this into `state_machine` and have some unit tests?

**Review by aslonnie — COMMENTED** [review]
I am a bit worried about these newly added stand-alone scripts that are not tested.

can we make the launching script as thin as possible so that we have better test coverage?

File: ci/ray_ci/automation/weekly_green_metric.py (hunk 1 above)
Comment (can-anyscale) [reply ci/ray_ci/automation/weekly_green_metric.py:34]: totally

**Comment by can-anyscale** [comment]
added unit tests

**Review by aslonnie — APPROVED** [review]

**Pull request merged**